drivers/i2c/i2c_mchp_sercom_g1.c

```diff
@@ -0,0 +1,1535 @@
+	data->target_mode = true;
+	i2c_target_runstandby_enable(dev);
+	i2c_target_enable(dev, true);
+	k_mutex_unlock(&data->i2c_bus_mutex);
+
+	return I2C_MCHP_SUCCESS;
+}
+
+static int i2c_mchp_target_unregister(const struct device *dev,
+				      struct i2c_target_config *target_cfg)
+{
+	struct i2c_mchp_dev_data *data = dev->data;
+
+	if (target_cfg == NULL) {
+		return -EINVAL;
+	}
+	if (data->target_mode != true) {
+		LOG_ERR("device are not configured as target device\n");
+		return -EBUSY;
+	}
+	if (data->target_config.address != target_cfg->address) {
+		return -EINVAL;
+	}
+
+	k_mutex_lock(&data->i2c_bus_mutex, K_FOREVER);
+	i2c_target_enable(dev, false);
+	i2c_target_int_disable(dev, SERCOM_I2CS_INTENSET_Msk);
+	i2c_reset_target_addr(dev);
+	data->target_mode = false;
+	data->target_config.address = 0x00;
+	data->target_config.callbacks = NULL;
+	i2c_target_enable(dev, true);
+	k_mutex_unlock(&data->i2c_bus_mutex);
+
+	return I2C_MCHP_SUCCESS;
+}
+#endif /*CONFIG_I2C_MCHP_TARGET*/
+
+#ifdef CONFIG_I2C_CALLBACK
+static int i2c_mchp_transfer_cb(const struct device *dev, struct i2c_msg *msgs, uint8_t num_msgs,
+				uint16_t addr, i2c_callback_t i2c_async_callback, void *user_data)
+{
+	struct i2c_mchp_dev_data *data = dev->data;
+	uint32_t addr_reg;
+
+	if (num_msgs == 0) {
+		return -EINVAL;
+	}
+	if (data->target_mode == true) {
+		LOG_ERR("Device currently running in target mode\n");
+		return -EBUSY;
+	}
+
+	for (uint8_t i = 0U; i < num_msgs; i++) {
+		if ((msgs[i].len == 0U) || (msgs[i].buf == NULL)) {
+			return -EINVAL;
+		}
+	}
+
+	k_mutex_lock(&data->i2c_bus_mutex, K_FOREVER);
+	data->num_msgs = num_msgs;
+	data->msgs_array = msgs;
+	data->i2c_async_callback = i2c_async_callback;
+	data->user_data = user_data;
+	data->target_addr = addr;
+	data->msg_index = 0;
+
+	i2c_controller_int_disable(dev, SERCOM_I2CM_INTENSET_Msk);
+	i2c_controller_int_flag_clear(dev, SERCOM_I2CM_INTFLAG_Msk);
+	i2c_controller_status_clear(dev, SERCOM_I2CM_STATUS_Msk);
+
+	data->current_msg.buffer = data->msgs_array[data->msg_index].buf;
+	data->current_msg.size = data->msgs_array[data->msg_index].len;
+	data->current_msg.status = 0;
+
+	addr_reg = addr << 1U;
+	bool is_read = ((data->msgs_array->flags & I2C_MSG_RW_MASK) == I2C_MSG_READ);
+
+	if (is_read == true) {
+		addr_reg |= I2C_MESSAGE_DIR_READ;
+	}
+	i2c_controller_addr_write(dev, addr_reg);
+	i2c_controller_int_enable(dev, SERCOM_I2CM_INTENSET_Msk);
+	k_mutex_unlock(&data->i2c_bus_mutex);
+
+	return I2C_MCHP_SUCCESS;
+}
+#endif /*CONFIG_I2C_CALLBACK*/
+
+#if !defined(CONFIG_I2C_MCHP_INTERRUPT_DRIVEN)
+static bool i2c_is_nack(const struct device *dev)
+{
+	bool retval;
+	sercom_registers_t *i2c_regs = ((const struct i2c_mchp_dev_config *)(dev)->config)->regs;
+
+	if ((i2c_regs->I2CM.SERCOM_CTRLA & SERCOM_I2CM_CTRLA_MODE_I2C_MASTER) ==
+	    SERCOM_I2CM_CTRLA_MODE_I2C_MASTER) {
+		if ((i2c_regs->I2CM.SERCOM_STATUS & SERCOM_I2CM_STATUS_RXNACK_Msk) ==
```

Comment:
can we remove if else condition
```suggestion
		retval = (i2c_regs->I2CM.SERCOM_STATUS & SERCOM_I2CM_STATUS_RXNACK_Msk) != 0;
        return retval;
```

Reply:
Updated